CorsixTH/Lua/humanoid_actions/seek_reception.lua

```diff
@@ -71,13 +74,14 @@ local function action_seek_reception_start(action, humanoid)
     local orientation = best_desk.object_type.orientations[best_desk.direction]
     local x = best_desk.tile_x + orientation.use_position[1]
     local y = best_desk.tile_y + orientation.use_position[2]
+    local dist = humanoid.world:getPathDistance(humanoid.tile_x, humanoid.tile_y, x, y)
     humanoid:updateDynamicInfo(_S.dynamic_info.patient.actions.on_my_way_to
       :format(best_desk.object_type.name))
     humanoid.waiting = nil
 
     -- We don't want patients which have just spawned to be joining the queue
     -- immediately, so walk them closer to the desk before joining the queue
-    if can_join_queue_at(humanoid, humanoid.tile_x, humanoid.tile_y) then
+    if can_join_queue_at(humanoid, humanoid.tile_x, humanoid.tile_y) and dist <= can_join_queue_distance then
```

Comment:
Your addition at all calls invalidates the name of the function. The function doesn't make the decision it claims to make any more since the additional condition could now block that decision. To clean up, I would suggest to either push the distance test inside the function, rename and possibly move the function so it doesn't make claims that aren't true, or even remove it entirely.

The function currently checks the humanoid for "inside the correct hospital". A function you'd expect as part of the humanoid code (it may even exist, didn't check) rather than part of reception.
Obviously "in the hospital" is better than joining at the spawn point so in that sense the function does what it claims compared to not having such a function, but you'd expect such a function to take the desk position into account, like you are adding. In other words, judging by the name of the the function it could also fit as part of the desk instance.

On the other hand, your addition makes a much tighter check, so arguably, "inside the right hospital" isn't very relevant any more. (With pathdistance, you also eliminated the option to walk outside near the desk, but still having to walk around the entire building.)

Reply:
Indeed Hospital:isInHospital plus a check for being in a room is all that can_join_queue_at is.

I agree with the can_join_queue_at with the extra parameter as thats an obvious alteration or break it down into the constituent calls, easier though to have just one function I guess. More I didn't integrate it in case there was something obvious I missed for now.

Outside you aren't in the hospital (well shouldn't be), but still possible within the distance, so can't join anyway before this distance check, the walk was truncated to within the bounds of the hospital already. But you can be in the hospital but not the same building when performing a SeekReception action so distance is important.

A patient can enter to join one and re-evaluate. But now they are on the otherside of the map. and choosing another desk and with the penalty in getUsageScores for a full queue, if you have more than one and one is full and one not, they might walk across the map entirely. You also don't want that QueueAction to start as they will queue from the other side of the map and lock the reception desk again, and given VIPs get priority. So either way its not ideal to rely on the hospital itself.

It maybe better to do the same best desk logic in the from outside the hospital and once they are in the hospital get nearest, but would probably want the TH feature of the queues being shared between reception desks placed close together before contemplating that.

Reply:
I always wondered why we don't treat reception just like a normal room, except for the door. It would allow all the queue management logic for rooms to be applied.

As for queue sharing, ideally they should be shared, but on the other hand, we don't share queues between several same  normal rooms close to each other either, eg 3 GPs next to each other.